### PR TITLE
fix(chartcuterie): query event when needed

### PR DESCRIPTION
**Pull request by sentaur-athena**
Fetching event might throw error so putting it where the try catch is. Also event is not needed till later.

### CODE REVIEW

**Review by cathteng — APPROVED** [review]

File: src/sentry/integrations/slack/message_builder/image_block_builder.py [hunk 1]
```diff
@@ -45,13 +44,13 @@ def _build_endpoint_regression_image_block(self) -> SlackBlock | None:
...
 
-        organization = self.group.organization
-        if self.event is None or self.event.transaction is None or self.event.occurrence is None:
-            return None
-        transaction_name = escape_transaction(self.event.transaction)
-        period = get_relative_time(anchor=get_approx_start_time(self.group), relative_days=14)
-
         try:
+            organization = self.group.organization
+            event = self.group.get_latest_event_for_environments()
```
Comment (cathteng) [inline src/sentry/integrations/slack/message_builder/image_block_builder.py:49]: in the future are we going to check for the type of group before doing this?

Comment (cathteng) [reply src/sentry/integrations/slack/message_builder/image_block_builder.py:49]: since this is what got us rate limited

Comment (sentaur-athena) [reply src/sentry/integrations/slack/message_builder/image_block_builder.py:49]: we are checking the type of group before this function in the `build_image_block` line 32.

Comment (cathteng) [reply src/sentry/integrations/slack/message_builder/image_block_builder.py:49]: oh i totally missed that, thanks!

**Pull request merged**